napari/layers/volume/tests/test_volume.py

```diff
@@ -0,0 +1,155 @@
+import numpy as np
+from vispy.color import Colormap
+from napari.layers import Volume
+from napari.util.colormaps.colormaps import TransFire, TransGrays
+
+
+def test_random_volume():
+    """Test instantiating Volume layer with random 3D data."""
+    shape = (10, 15, 20)
+    np.random.seed(0)
+    data = np.random.random(shape)
+    layer = Volume(data)
+    assert np.all(layer.data == data)
+    assert layer.ndim == len(shape)
+    assert layer.shape == shape
+    assert layer.range == tuple((0, m, 1) for m in shape)
+    assert layer._data_view.shape == shape[-3:]
+
+
+def test_all_zeros_volume():
+    """Test instantiating Volume layer with all zeros data."""
+    shape = (10, 15, 20)
+    data = np.zeros(shape, dtype=float)
+    layer = Volume(data)
+    assert np.all(layer.data == data)
+    assert layer.ndim == len(shape)
+    assert layer.shape == shape
+    assert layer._data_view.shape == shape[-3:]
+
+
+def test_integer_volume():
+    """Test instantiating Volume layer with integer data."""
+    shape = (10, 15, 20)
+    np.random.seed(0)
+    data = np.round(10 * np.random.random(shape)).astype(int)
+    layer = Volume(data)
+    assert np.all(layer.data == data)
+    assert layer.ndim == len(shape)
+    assert layer.shape == shape
+    assert layer._data_view.shape == shape[-3:]
+
+
+def test_3D_volume():
```

Comment:
maybe make this test a `4D` volume

Reply:
why? vispy doesn't allow visualization of rgb, rgba volumes, right?

Reply:
if I test a 4D volume I will get the below error

``` python     self.set_data(vol, clim)
  File "/usr/local/lib/python3.6/site-packages/vispy/visuals/volume.py", line 465, in set_data
    raise ValueError('Volume visual needs a 3D image.')
ValueError: Volume visual needs a 3D image.```

Reply:
You need to slice the array so that you only pass a 3D array to vispy - I've left comments about this in the volume layer. I'll explain more there. This is to support volumetric timeseries data for example

Reply:
added that and pushing